src/service.c

```diff
@@ -1062,6 +1073,20 @@ static void on_custom_activated (GSimpleAction *pAction G_GNUC_UNUSED, GVariant
     ayatana_common_utils_open_url (sUri);
 }
 
+static void onClose (GSimpleAction *pAction G_GNUC_UNUSED, GVariant *pParam G_GNUC_UNUSED, gpointer pUserData)
+{
+    gboolean bSuspendable = rda_session_can_be_suspended ();
+
+    if (bSuspendable)
+    {
+        rda_session_suspend ();
+    }
+    else
+    {
+        rda_session_terminate ();
```

Comment:
rda_session_terminate is basically like a logout. So, if you may, you can connect ``rda_session_terminate()`` to the logout menu item. The rda_session_terminate will call the remote technologies terminate session functionality and it might do a better cleanup (or a more enforced cleanup) then a simple session logout. For X2Go, both work well, only the rda_session_terminate calls x2goterminate-session which also kills the underlying remote Xserver (x2goagent, x2gokdrive).

Reply:
However, the underlying Xserver will also disappear once the session is not running anymore.